homeassistant/components/backup/config.py

```diff
@@ -135,6 +146,80 @@ async def update(
         self.save()
 
 
+@dataclass(kw_only=True)
+class DeleteAfterConfig:
+    """Represent the delete after configuration."""
+
+    copies: int | None = None
+    days: int | None = None
+
+    def apply(self, manager: BackupManager) -> None:
+        """Apply delete after configuration."""
+        if self.days is not None:
+            self._schedule_next(manager)
+        else:
+            self._unschedule_next(manager)
+
+    def to_dict(self) -> StoredDeleteAfterConfig:
+        """Convert delete after configuration to a dict."""
+        return StoredDeleteAfterConfig(
+            copies=self.copies,
+            days=self.days,
+        )
+
+    @callback
+    def _schedule_next(
+        self,
+        manager: BackupManager,
+    ) -> None:
+        """Schedule the next delete after days."""
+        self._unschedule_next(manager)
+
+        async def _delete_backups(now: datetime) -> None:
+            """Delete backups older than days."""
+            self._schedule_next(manager)
+
+            def _backups_filter(backups: dict[str, Backup]) -> dict[str, Backup]:
+                """Return backups older than days to delete."""
+                if self.days is None:
+                    return {}
+                now = dt_util.now()  # local time
+                return {
+                    backup_id: backup
+                    for backup_id, backup in backups.items()
+                    if dt_util.parse_datetime(backup.date, raise_on_error=True)
```

Comment:
It's a bit silly to have to parse the time stamps, I think we should update the dataclass to store a datetime object?